pkg/state/state_skel.go

```diff
@@ -112,6 +112,14 @@ func (s *stateSkel) createOrUpdateObjs(
 			return err
 		}
 
+		// Set resource version
```

Comment:
So, i think this deserves some more info in the comment.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency

Reply:
From the doc this value must be passed unmodified back to the server and the  only way for a client to know the expected value of resourceVersion is to have received it from the server

Reply:
It is used by the kubernetes API server to implement optimistic concurrency for PUT operations.
id like you to add a comment here on why we are setting the resource version so its clear.